### PR TITLE
Bug Fix

### PR DESCRIPTION
**Pull request by bigfluffycookie**
fixes https://github.com/SonarSource/sonarlint-visualstudio/issues/3701

### CODE REVIEW

File: src/Integration.Vsix/Analysis/AnalysisConfigMonitor.cs [hunk 1]
```diff
@@ -75,15 +71,8 @@ internal sealed class AnalysisConfigMonitor : IAnalysisConfigMonitor, IDisposabl
...
-            if (activeSolutionBoundTracker.CurrentConfiguration.Mode == SonarLintMode.Standalone)
-            {
-                logger.WriteLine(AnalysisStrings.ConfigMonitor_UserSettingsChanged);
-                OnSettingsChangedAsync().Forget();
-            }
-            else
-            {
-                logger.WriteLine(AnalysisStrings.ConfigMonitor_UserSettingsIgnoredForConnectedModeLanguages);
-            }
+            logger.WriteLine(AnalysisStrings.ConfigMonitor_UserSettingsChanged);
```
Comment (bigfluffycookie) [inline src/Integration.Vsix/Analysis/AnalysisConfigMonitor.cs:74]: I think its fine to leave this message as if user specified settings are not supported it'll print this later "[RuleSettings] Using connected mode settings. User-specified settings in settings.json will be ignored."

File: src/Integration.Vsix.UnitTests/Analysis/AnalysisConfigMonitorTests.cs [hunk 2]
```diff
@@ -19,13 +19,11 @@
  */
```
[the remark this answers is not in the document]
Comment (duncanp-sonar) [reply src/Integration.Vsix.UnitTests/Analysis/AnalysisConfigMonitorTests.cs:20]: The best kind of change...

File: src/Integration.Vsix/Analysis/AnalysisConfigMonitor.cs [hunk 3]
```diff
@@ -75,15 +71,8 @@ internal sealed class AnalysisConfigMonitor : IAnalysisConfigMonitor, IDisposabl
...
         private void OnUserSettingsChanged(object sender, EventArgs e)
         {
-            if (activeSolutionBoundTracker.CurrentConfiguration.Mode == SonarLintMode.Standalone)
-            {
-                logger.WriteLine(AnalysisStrings.ConfigMonitor_UserSettingsChanged);
-                OnSettingsChangedAsync().Forget();
-            }
-            else
-            {
-                logger.WriteLine(AnalysisStrings.ConfigMonitor_UserSettingsIgnoredForConnectedModeLanguages);
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/Analysis/AnalysisConfigMonitor.cs:85]: Is this resource string still used?

File: src/Integration.Vsix/Analysis/AnalysisConfigMonitor.cs (hunk 1 above)
Comment (duncanp-sonar) [inline src/Integration.Vsix/Analysis/AnalysisConfigMonitor.cs:74]: I think it's worth adding a comment explaining why we might trigger re-analysis in Connected Mode e.g. 

> There is a corner-case where we want to raise the the event even in Connected Mode - see #3701.

**Review by duncanp-sonar — APPROVED** [review]
Nice. Just comments about comments.

**Comment by duncanp-sonar** [comment]
@bigfluffycookie I tested the fix and it works as expected. @

**Pull request merged**